### ISSUE TITLE
salt-run list jobs target returns unknown-target

### ISSUE THREAD

**Issue by tuxthepenguin84** [saltstack/salt#51375]
### Description of Issue/Question
When running the following almost all the jobs returned have "unknown-target" listed as the Targets
```
salt-run jobs.list_jobs search_function='state.apply'
```
This issue seems identical to the following:
https://github.com/saltstack/salt/issues/24140
https://github.com/saltstack/salt/issues/48734

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)
Clean install of 2018.3.3 through BOOTSTRAP - MULTI-PLATFORM
https://repo.saltstack.com/#bootstrap

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)
```
salt '*' state.apply samba
salt-run jobs.list_jobs search_function='state.apply'

20190128154817549668:
    ----------
    Arguments:
    Function:
        state.apply
    StartTime:
        2019, Jan 28 15:48:17.549668
    Target:
        unknown-target
    Target-type:
        list
    User:
        root
20190128155950770880:
    ----------
    Arguments:
    Function:
        state.apply
    StartTime:
        2019, Jan 28 15:59:50.770880
    Target:
        unknown-target
    Target-type:
        list
    User:
        root
20190128162534350695:
    ----------
    Arguments:
    Function:
        state.apply
    StartTime:
        2019, Jan 28 16:25:34.350695
    Target:
        unknown-target
    Target-type:
        list
    User:
        root

```

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
```
Salt Version:
           Salt: 2018.3.3

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 12 2018, 14:36:49)
   python-gnupg: 0.3.8
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-141-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

**Comment by garethgreenaway** [comment]
@tuxthepenguin84 Thanks for the report.  The fix from PR https://github.com/saltstack/salt/pull/50328 was made after 2018.3.3 was released, so it will be available in the next 2018.3 release.

**Comment by tuxthepenguin84** [comment]
Thank you, I applied the fix from PR #50328 and this is has been resolved for me.
